File: src/2140. Solving Questions With Brainpower.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& qestions) {
        long long dp[200001] = {};
        for (int i = qestions.size() - 1; i >= 0; --i) {
            dp[i] = max(dp[i + 1], dp[qestions[i][1] + i + 1] + qestions[i][0]);
        }
        return dp[0];
    }
};

File: src/2140. Solving Questions With Brainpower.py
class Solution:
    def mostPoints(self, questions: List[List[int]]) -> int:
        dp = [0] * 200001
        for i in range(len(questions) - 1, -1, -1):
            dp[i] = max(dp[i + 1], dp[questions[i][1] + i + 1] + questions[i][0])
        return dp[0]